src/deepsparse/transformers/schemas/text_generation_schemas.py

```diff
@@ -22,20 +23,21 @@
 
 
 # Based off of https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig # noqa E501
+@dataclass
 class GenerationDefaults:
     # Parameters that control the length of the output
-    max_length = None
-    max_new_tokens = 100
+    max_length: int = field(default=None)
+    max_new_tokens: int = field(default=100)
     # Parameters that control the generation strategy used
-    do_sample = False
+    do_sample: bool = field(default=False)
     # Parameters for manipulation of the model output logits
-    temperature = 1.0
-    top_k = 50
-    top_p = 1.0
-    repetition_penalty = 1.0
+    temperature: float = field(default=1.0)
+    top_k: int = field(default=50)
```

Comment:
Pydantic Fields would go in nicely here, specially cause we can provide a description of what each argument is

Reply:
doesnt seem like it needs to be a pydantic model if only used internally and just holds default data